toolchain/lower/file_context.cpp

```diff
@@ -373,10 +372,15 @@ auto FileContext::BuildDISubprogram(const SemIR::Function& function,
   auto loc = converter_.ConvertLoc(
       function.definition_id,
       [](DiagnosticLoc, const Internal::DiagnosticBase<>&) {});
-  // FIXME: Add more details here, including mangled name, real subroutine type
-  // (once type information is built), etc.
+  auto opt_name = sem_ir().names().GetAsStringIfIdentifier(function.name_id);
+  CARBON_CHECK(opt_name) << "Unexpected special name for function: "
+                         << function.name_id;
+  llvm::StringRef name = *opt_name;
+  llvm::StringRef linkage_name = llvm_function->getName();
```

Comment:
Is it still worth having separate variables for these?

Reply:
Oh, rather than rolling the expressions into their use? Sure - done!